ptbcontrib/roles/roleshandler.py

```diff
@@ -0,0 +1,88 @@
+#!/usr/bin/env python
+#
+# A library containing community-based extension for the python-telegram-bot library
+# Copyright (C) 2020
+# The ptbcontrib developers
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser Public License for more details.
+#
+# You should have received a copy of the GNU Lesser Public License
+# along with this program.  If not, see [http://www.gnu.org/licenses/].
+"""This module contains the RoleHandler class."""
+from typing import Any
+
+from telegram import Update
+from telegram.ext import Handler, CallbackContext, Dispatcher
+
+from .roles import Role, Roles
+
+BOT_DATA_KEY: str = 'ptbcontrib_roles_bot_data_key'
+""":obj:`str`: The key used to store roles in ``bot_data``."""
+
+
+def setup_roles(dispatcher: Dispatcher) -> Roles:
+    """
+    Retrieves the :class:`ptbcontrib.roles.Roles` instance stored in :attr:`dispatcher.bot_data`
+    or creates a new one and saves it under :attr:`BOT_DATA_KEY`.
+
+    Args:
+        dispatcher (:class:`telegram.ext.Dispatcher`): The dispatcher
+
+    Returns:
+        The :class:`ptbcontrib.roles.Roles` instance to be used.
+    """
+    if BOT_DATA_KEY not in dispatcher.bot_data:
+        dispatcher.bot_data[BOT_DATA_KEY] = Roles(dispatcher.bot)
+    return dispatcher.bot_data[BOT_DATA_KEY]
```

Comment:
```suggestion
    return dispatcher.bot_data.setdefault(BOT_DATA_KEY, Roles(dispatcher.bot))
```
(if instantiating `Roles` is cheap)